app/controllers/application_controller.rb

```diff
@@ -62,4 +62,8 @@ def render_unauthorized
     headers['WWW-Authenticate'] = 'Token realm="Application"'
     render json: 'Not Authorized', status: 401
   end
+
+  rescue_from ActionController::ParameterMissing do
+    render json: { error: 'parameter missing' }, status: :bad_request
+  end
```

Comment:
@austin-martinez I added this to the base controller to return a 400 when parameters are missing. I thought this was novel enough that I should ask before merging it.